pkg/cidata/cidata.go

```diff
@@ -50,17 +61,26 @@ func setupEnv(y *limayaml.LimaYAML) (map[string]string, error) {
 			}
 		}
 	}
-	// Replace "localhost" in proxy settings with the gateway address
-	localhostRegexes := []*regexp.Regexp{
-		regexp.MustCompile(`\blocalhost\b`),
-		regexp.MustCompile(`\b127.0.0.1\b`),
-	}
+	// Replace IP that IsLoopback in proxy settings with the gateway address
 	for _, name := range append(lowerVars, upperVars...) {
 		value, ok := env[name]
 		if ok && !strings.EqualFold(name, "no_proxy") {
-			for _, re := range localhostRegexes {
-				value = re.ReplaceAllString(value, qemu.SlirpGateway)
+			u, err := url.Parse(value)
+			if err != nil {
+				logrus.Warnf("Ignoring invalid proxy %q=%v: %s", name, value, err)
+				continue
+			}
+
+			for _, ip := range netLookupIP(u.Hostname()) {
+				newHost := qemu.SlirpGateway
+				if u.Port() != "" {
+					newHost = net.JoinHostPort(newHost, u.Port())
+				}
+				if ip.IsLoopback() {
+					u.Host = newHost
+				}
```

Comment:
It doesn't really matter, but all this code could be inside the `IsLoopback` conditional:

```suggestion
				if ip.IsLoopback() {
					newHost := qemu.SlirpGateway
					if u.Port() != "" {
						newHost = net.JoinHostPort(newHost, u.Port())
					}
					u.Host = newHost
				}
```